code/modules/mob/living/bloodcrawl.dm

```diff
@@ -37,58 +40,92 @@
-					src << "<span class='warning'><b>AAH! THEIR FLESH! IT BURNS!</b></span>"
-					adjustBruteLoss(25) //I can't use adjustHealth() here because bloodcrawl affects /mob/living and adjustHealth() only affects simple mobs
-					kidnapped.loc = get_turf(B)
-					kidnapped.visible_message("<span class='warning'>[B] violently expels [kidnapped]!</span>")
-				else
-					src << "<span class='danger'>You devour [kidnapped]. Your health is fully restored.</span>"
-					src.adjustBruteLoss(-1000)
-					src.adjustFireLoss(-1000)
-					src.adjustOxyLoss(-1000)
-					src.adjustToxLoss(-1000)
-					kidnapped.ghostize()
-					qdel(kidnapped)
-			else
-				src << "<span class='danger'>You happily devour... nothing? Your meal vanished at some point!</span>"
-		src.notransform = 0
+		bloodpool_sink(B)
+		src.notransform = FALSE
 	return 1
 
+/mob/living/proc/bloodpool_sink(obj/effect/decal/cleanable/B)
+	var/turf/mobloc = get_turf(src.loc)
+
+	src.visible_message("<span class='warning'>[src] sinks into the pool of blood!</span>")
+	playsound(get_turf(src), 'sound/magic/enter_blood.ogg', 100, 1, -1)
+	// Extinguish, unbuckle, stop being pulled, set our location into the
+	// dummy object
+	var/obj/effect/dummy/slaughter/holder = PoolOrNew(/obj/effect/dummy/slaughter,mobloc)
+	src.ExtinguishMob()
+	if(buckled)
+		buckled.unbuckle_mob(src,force=1)
+	if(buckled_mobs.len)
+		unbuckle_all_mobs(force=1)
+	if(pulledby)
+		pulledby.stop_pulling()
+	src.loc = holder
+	src.holder = holder
+
+	// if we're not pulling anyone, or we can't eat anyone
+	if(!src.pulling || src.bloodcrawl != BLOODCRAWL_EAT)
+		return
+
+	// if the thing we're pulling isn't alive
+	if (!(istype(src.pulling, /mob/living)))
+		return
+
+	var/mob/living/victim = src.pulling
+	var/kidnapped = FALSE
+
+	if(victim.stat == CONSCIOUS)
+		src.visible_message("<span class='warning'>[victim] kicks free of the blood pool just before entering it!</span>", null, "<span class='notice'>You hear splashing and struggling.</span>")
+	else if(victim.reagents && victim.reagents.has_reagent("demonsblood"))
+		visible_message("<span class='warning'>Something prevents [victim] from entering the pool!</span>", "<span class='warning'>A strange force is blocking [victim] from entering!</span>", "<span class='notice'>You hear a splash and a thud.</span>")
+	else
+		victim.loc = src
+		victim.emote("scream")
+		src.visible_message("<span class='warning'><b>[src] drags [victim] into the pool of blood!</b></span>", null, "<span class='notice'>You hear a splash.</span>")
+		kidnapped = TRUE
+
+	if(kidnapped)
+		var/success = bloodcrawl_consume(B, victim)
+		if(!success)
+			src << "<span class='danger'>You happily devour... nothing? Your meal vanished at some point!</span>"
+	return 1
+
+/mob/living/proc/bloodcrawl_consume(obj/effect/decal/cleanable/B, mob/living/victim)
+	src << "<span class='danger'>You begin to feast on [victim]. You can not move while you are doing this.</span>"
+
+	for(var/i = 3; i > 0; i--)
```

Comment:
literally not my code, just copying what was there before. Given all it does is loop the sound code 3 times, I'll just have a normal for loop counting up